### ISSUE TITLE
[accessibility issue] VariableInput fields do not have a programmatic name associated with them

### ISSUE THREAD

**Issue by ckbedwell** [grafana/grafana#79517]
**Steps to reproduce**:
Go to a Grafana dashboard with dropdown select variables. Activate the button so the dropdown and input field appears. 

**Actual Result**:
The input is not programmatically associated with the accompanying label as it has no id attribute.

**Expected Result**
The input should be associated with the label.

![A screenshot of a Grafana dashboard with a variable select. VoiceOver's caption box is also present and Chrome devtools. They demonstrate that the input field is not associated with its label.](https://github.com/grafana/grafana/assets/6906380/c847b31d-b10f-4fef-9eb6-6a3970de1667)

UHG Audit ID: 1.31121

**Comment by tskarhed** [comment]
I think I see the issue here. The `id` is actually on the list of items. We should look into if it is possible to use the `<datalist>` element for this.
